packages/uui-color-picker/lib/uui-color-picker.element.ts

```diff
@@ -382,15 +382,25 @@ export class UUIColorPickerElement extends LabelMixin('label', LitElement) {
   }
 
   setColor(colorString: string | HslaColor) {
-    this._colord = new Colord(colorString);
 
-    const { h, l, s, a } = this._colord.toHsl();
+    const colord = new Colord(colorString);
+
+    const { h, s, l, a } = colord.toHsl();
 
     this.hue = h;
     this.saturation = s;
     this.lightness = l;
     this.alpha = this.opacity ? a * 100 : 100;
 
+    const hslaColor = colorString as HslaColor;
+
+    // Workaround as hue isn't correct after changing hue slider, but Colord parse hue value as zero when color is black.
+    if (hslaColor && hslaColor.h) {
+      this.hue = hslaColor.h;
+    }
```

Comment:
It had an incorrect parsing from `HslaColor` object:
https://github.com/omgovich/colord/issues/116